smartsnippets/templatetags/smartsnippets_tags.py

```diff
@@ -61,10 +63,41 @@ def split(string_to_split, delimiter=None):
 def as_dict(**kwargs):
     return OrderedDict(kwargs)
 
+
 @register.assignment_tag
 def current_timestamp():
     return datetime.utcnow().strftime("%Y%m%d_%H%M%S_%f")
 
+
 @register.assignment_tag(takes_context=True)
-def from_context(context, name):
-    return context.get(name, None)
\ No newline at end of file
+def from_context(context, name, sep=None, empty=None):
+    result_items = [
+        (context.get(n, None) or empty)
+        for n in name.split(sep or ',')]
+    return result_items if len(result_items) > 1 else result_items[0]
+
+
+def _by_args(args):
```

Comment:
I would prefer if this method would be named more explicitly like *_select_by_args*. 
Add some docstrings.
Also, you can use *operator.itegetter* and *operator.attrgetter*. These return functions.